libtelemetry/src/main/java/com/mapbox/android/telemetry/NavigationMetadata.java

```diff
@@ -45,17 +45,25 @@
   private Boolean batteryPluggedIn;
   private Integer batteryLevel;
   private String connectivity;
+  private String tripIdentifier;
+  private Integer legIndex;
+  private Integer legCount;
+  private Integer stepIndex;
+  private Integer voiceIndex = null;
+  private Integer bannerIndex = null;
+  private Integer totalStepCount;
 
   public NavigationMetadata(Date startTimestamp, int distanceCompleted, int distanceRemaining, int durationRemaining,
```

Comment:
It looks like the constructor became quite unmaintainable. Any thoughts on how we can improve on readability? Perhaps consider Builders?

Reply:
If I remember correctly, we purposefully moved away from Builders when we came up with the new implementation at 3.0.0. Our goal was to benefit developers. The good news is this can be moved out of the sdk during the refactor and external sdks can change it up or handle this as they want to.

Reply:
With that huge number of mandatory/optional parameters should we consider reverting back to builders? public constructors with many parameters of the same type are super error prone.